src/gx/CGxDevice.hpp

```diff
@@ -120,6 +120,7 @@ class CGxDevice {
         virtual void PoolSizeSet(CGxPool*, uint32_t) = 0;
         virtual char* BufLock(CGxBuf*);
         virtual int32_t BufUnlock(CGxBuf*, uint32_t);
+        virtual void BufData(CGxBuf* buf, void* data, size_t size, uintptr_t offset);
```

Comment:
Sorry, missed this earlier: I believe `data` here is `const void*`. Other than that, I think this is ready to merge!